src/main/java/lily/Parser.java

```diff
@@ -0,0 +1,165 @@
+package lily;
+
+import lily.task.LilyException;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Makes sense of commands input by the user and executes them.
+ * 
+ * @author Hong Yi En, Ian
+ * @version Feb 2022 (AY21/22 Sem 2)
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+    private Storage st;
+    private String sentence;
+    private String[] parsedSentence;
+    private String command;
+
+    /**
+     * Creates a new Parser Object.
+     * 
+     * @param t The TaskList referred to.
+     * @param ui The UI needed to interact with the user.
+     * @param st The Storage needed to store data.
+     */
+    public Parser(TaskList t, Ui ui, Storage st) {
+        this.tasks = t;
+        this.ui = ui;
+        this.st = st;
+    }
+
+
+    /**
+     * The main loop of interacting with the user.
+     * Decodes the input and calls the relevant functions.
+     */
+    public void parse() {
+        userInteracting: while (true) {
+            try {
+                sentence = ui.readCommand();
+                parsedSentence = sentence.split(" ");
+                command = parsedSentence[0];
+                switch (command) {
+                case "bye":
+                    st.save(tasks);
+                    break userInteracting;
+    
+                case "list":
+                    ui.showList(tasks);
+                    break;
+    
+                case "mark":
+                    int addIdx = Integer.parseInt(parsedSentence[1]); // base 1
+                    ui.showMarked(tasks.mark(addIdx - 1), addIdx);
+                    break;
+    
+                case "unmark":
+                    int delIdx = Integer.parseInt(parsedSentence[1]);
+                    ui.showUnmarked(tasks.unmark(delIdx - 1), delIdx);
+                    break;
+    
+                case "todo":
+                    ui.showTaskAdded(tasks.addTodo(findTodoDescStart(sentence)), tasks.getSize());
```

Comment:
Nice, clean use of switch-case with appropriate spacing to parse commands

Reply:
Thanks!